File: LeetCode - May/MaxCircularSubArray.java
class Solution {
    public int maxSubarraySumCircular(int[] nums) {
        
        int maxcurrent = nums[0];
        int maxsofar = nums[0];
        int mincurrent = nums[0];
        int minsofar = nums[0];
        for(int i=1; i<nums.length; i++){
            maxcurrent = Math.max(nums[i], maxcurrent+nums[i]);
            maxsofar =Math.max(maxsofar, maxcurrent);
            mincurrent = Math.min(nums[i], mincurrent+nums[i]);
            minsofar = Math.min(minsofar, mincurrent);
        }
        int sum = Arrays.stream(nums).sum();
        if(minsofar == sum)
            return maxsofar;
        else
            return Math.max(maxsofar,sum-minsofar);
    }
}

File: LeetCode - May/MaxCircularSubArray.py
class Solution:
    def maxSubarraySumCircular(self, nums: List[int]) -> int:
        maxcurrent = nums[0]
        maxsofar = nums[0]
        mincurrent = nums[0]
        minsofar = nums[0]

        for i in range(1,len(nums)):
            maxcurrent = max(nums[i], maxcurrent+nums[i])
            maxsofar = max(maxcurrent, maxsofar)
            mincurrent = min(nums[i], mincurrent+nums[i])
            minsofar = min(mincurrent, minsofar)
        
        if minsofar == sum(nums):
            return maxsofar
        else:
            return max(maxsofar,sum(nums)-minsofar)
        
        